### PR TITLE
Web storage Review

### PR DESCRIPTION
**Pull request by Felipeg005**
Implemented the following interactions:

1. When the user changes the content of any input field, the data is saved to the local storage.
2. When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.

The following data model was used:

A single JavaScript object contains all the data from the entire form and saves it in local storage.

Thank you for your review :smile: .

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -243,8 +246,37 @@ function formLowerCaseValidate(submitForm) {
...
+}
+
+function preFillFields() {
+  const storageForm = JSON.parse(localStorage.getItem('formInfo'));
+  if (storageForm) {
+    nameField.value = storageForm.name;
+    mailField.value = storageForm.email;
+    messageField.value = storageForm.message;
+  }
+}
```
Comment (Davidosky007) [inline script.js:267]: * Kindly implement these functionalities as it is required
* The user-entered value be saved in local storage without clicking the submit button
* When the user changes the content of any input field, the data is saved to the local storage 
* Kindly 🙏 fix this as currently, the user has to click submit for input data to be stored in local storage and when changes are made on the input values the local storage isn't updated. [the specific Project requirement](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/10_localstorage.md#:~:text=When%20the%20user%20changes%20the%20content%20of%20any%20input%20field%2C%20the%20data%20is%20saved%20to%20the%20local%20storage.)

File: script.js [hunk 2]
```diff
@@ -243,8 +246,37 @@ function formLowerCaseValidate(submitForm) {
   }
 }
 
+function storageEmailData() {
```
Comment (Davidosky007) [inline script.js:248]: * Kindly fix this error which shows on the console on page load 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
# Hi @Felipeg005 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️\

## _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: script.js (hunk 2 above)
Comment (Felipeg005) [reply script.js:248]: Thank you Davidosky007 for your review :smile:, you are right, I wrote the code to save data in local storage when submit event occurs...

In order to improve the code next actions were taken:

1. Change functionalities so the value it's saved in local storage without clicking the submit button.
2. When the user changes the content of any input field, the data is saved to the local storage instantly.
3. The console error it's fixed.
image

All the required changes are updated.

**Review by hemantso — APPROVED** [review]
Hello team, 

Well done on making the requested changes. 🚀 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 

Alt Text

Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**